File: src/js/main/core/RichJsonCommandHolder.js
import {mergeIntoWithoutRebind} from "../helper/RichJsonHelper.js";
import {__executeRefCommand} from "../commands/RichJson_ref.js";
import {__executeEnvCommand} from "../commands/RichJson_env.js";
import {__executeFileCommand} from "../commands/RichJson_file.js";
import {__executeFolderCommand} from "../commands/RichJson_folder.js";
import {__executeMergeCommand} from "../commands/RichJson_merge.js";
import {__executeMergeFolderCommand} from "../commands/RichJson_merge_folder.js";
import {__executeCopyCommand} from "../commands/RichJson_copy.js";
import {__executeCloneCommand} from "../commands/RichJson_clone.js";
import {__executeThisCommand} from "../commands/RichJson_this.js";
import {__executeInvokeCommand} from "../commands/RichJson_invoke.js";
import {__RICH_JSON_CONFIG} from "../other/RichJsonConfiguration.js";

class RichJsonCommandHolder {
    void = function () {
    }
    available = {
        ref: __executeRefCommand,
        env: __executeEnvCommand,
        this: __executeThisCommand,

        merge: __executeMergeCommand,
        copy: __executeCopyCommand,
        clone: __executeCloneCommand,
        invoke: __executeInvokeCommand,

        file: __executeFileCommand,
        folder: __executeFolderCommand,
        merge_folder: __executeMergeFolderCommand,
    };
    enabled = {};
    built_in = {};
    kcmd_ignored = {};

    constructor() {
        mergeIntoWithoutRebind(this.enabled, this.available);
        mergeIntoWithoutRebind(this.built_in, this.available);
    }
}

export let __RICH_JSON_COMMANDS = new RichJsonCommandHolder();
export let __RICH_JSON_LATE_APPLIES = ["this", "clone", "invoke"];


/**
 * Toggles a RichJson command's enabled state.
 * @param {string} command - Name of the command to toggle.
 * @param {boolean} enabled - Whether to enable or disable the command.
 * @throws {Error} If the command is not in the available registry.
 */
export function setCommandEnabled(command, enabled) {
    if (!Object.hasOwn(__RICH_JSON_COMMANDS.available, command)) {
        __throwCommandNotFound(command);
    }

    if (enabled) {
        __RICH_JSON_COMMANDS.enabled[command] = __RICH_JSON_COMMANDS.available[command];
    } else {
        __RICH_JSON_COMMANDS.enabled[command] = __RICH_JSON_COMMANDS.void;
    }

    if (__RICH_JSON_CONFIG.debugEnabled) {
        console.debug(`RichJson command '${command}' was ${enabled ? "enabled" : "disabled"}.`);
    }
}

export function __throwCommandNotFound(command) {
    throw (`RichJson Command '${command}' not found`);
}

File: src/python/main/core/rich_json_command_holder.py
from ..helper.rich_json_helper import merge_into_without_rebind

class RichJsonCommandHolder:
    """
    Holds and manages the available, enabled, and built-in RichJson commands.
    """
    def __init__(self):
        from ..commands.rich_json_ref import _execute_ref_command
        from ..commands.rich_json_env import _execute_env_command
        from ..commands.rich_json_file import _execute_file_command
        from ..commands.rich_json_folder import _execute_folder_command
        from ..commands.rich_json_merge import _execute_merge_command
        from ..commands.rich_json_merge_folder import _execute_merge_folder_command
        from ..commands.rich_json_copy import _execute_copy_command
        from ..commands.rich_json_clone import _execute_clone_command
        from ..commands.rich_json_this import _execute_this_command
        from ..commands.rich_json_invoke import _execute_invoke_command

        self.void_command = lambda *args, **kwargs: None

        self.available = {
            "ref": _execute_ref_command,
            "env": _execute_env_command,
            "this": _execute_this_command,
            "merge": _execute_merge_command,
            "copy": _execute_copy_command,
            "clone": _execute_clone_command,
            "invoke": _execute_invoke_command,
            "file": _execute_file_command,
            "folder": _execute_folder_command,
            "merge_folder": _execute_merge_folder_command,
        }

        self.enabled = {}
        self.built_in = {}
        self.kcmd_ignored = {}

        merge_into_without_rebind(self.enabled, self.available)
        merge_into_without_rebind(self.built_in, self.available)